tests/Microsoft.Bot.Builder.Prompts.Tests/NumberPromptTests.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Adapters;
+using Microsoft.Bot.Builder.Middleware;
+using Microsoft.Bot.Builder.Storage;
+using Microsoft.Bot.Schema;
+using Microsoft.Recognizers.Text;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Bot.Builder.Prompts.Tests
+{
+    public class TestState : StoreItem
+    {
+        public bool InPrompt { get; set; } = false;
+    }
+
+    [TestClass]
+    [TestCategory("Prompts")]
+    [TestCategory("Number Prompts")]
+    public class NumberPromptTests
+    {
+        [TestMethod]
+        public async Task NumberPrompt_Float()
+        {
+            TestAdapter adapter = new TestAdapter()
+                .Use(new ConversationState<TestState>(new MemoryStorage()));
+
+            await new TestFlow(adapter, async (context) =>
+                {
+                    var state = ConversationState<TestState>.Get(context);
+                    var numberPrompt = new NumberPrompt<float>(Culture.English);
+                    if (!state.InPrompt)
+                    {
+                        state.InPrompt = true;
+                        await numberPrompt.Prompt(context, "Gimme:");
+                    }
+                    else
+                    {
+                        var result = await numberPrompt.Recognize(context);
+                        if (result == null)
```

Comment:
The only thing I don't like about these prompts is that Null Checking is required to use them. Is there a clear way to get away from that? The Tuple approach I had seemed much clearer, and the RecognizeResult.DidPass approach that Brandon suggested also seemed valid.

Reply:
again, `PromptResult<T>` would solve this.